cpp/src/streaming/cudf/table_chunk.cpp

```diff
@@ -176,6 +178,22 @@ TableChunk TableChunk::copy(MemoryReservation& reservation) const {
                         std::move(packed_columns.metadata),
                         br->move(std::move(packed_columns.gpu_data), stream())
                     );
+
+                    // Handle the case where `cudf::pack` allocates slightly more than the
+                    // input size. This can occur because cudf uses aligned allocations,
+                    // which may exceed the requested size. To accommodate this, we
+                    // slightly increase the reservation if the packed data fits within
+                    // the aligned size.
+                    if (packed_data->data->size > reservation.size()) {
+                        auto const aligned_size = rmm::align_up(reservation.size(), 1024);
+                        if (packed_data->data->size <= aligned_size) {
+                            reservation =
+                                br->reserve(
+                                      MemoryType::HOST, packed_data->data->size, true
```

Comment:
We allow overbooking here because we know we have a reservation for _nearly_ the amount we need, so we're just "this should be fine".

Should we extend the `reserve` API to allow it to consume a given reservation. i.e. here what we're trying to do is add (up to) 1023 bytes to the existing reservation. It seems like a "truer" thing to do would be to `reservation.resize(...)`